charts/px-central/templates/px-backup/pre-upgrade-hook/pre-upgrade-check.yaml

```diff
@@ -0,0 +1,205 @@
+  template:
+    spec:
+      initContainers:
+        - name: init-log-permissions
+          image: busybox
+          command: ["sh", "-c", "mkdir -p /logs && chmod -R 777 /logs"]
+          volumeMounts:
+          - name: log-volume
+            mountPath: /logs
+        - name: init-dump-permissions
+          image: busybox
+          command: ["sh", "-c", "mkdir -p /dumps && chmod -R 777 /dumps"]
+          volumeMounts:
+          - name: dump-volume
+            mountPath: /dumps
+      containers:
+        - name: check-upgrade
+          image: {{ printf "%s/%s/%s:%s" .Values.images.preUpgradeHookImage.registry .Values.images.preUpgradeHookImage.repo .Values.images.preUpgradeHookImage.imageName .Values.images.preUpgradeHookImage.tag }}
+          imagePullPolicy: {{ .Values.images.pullPolicy }}
+          command: ["python", "/pxcentral-pre-upgrade/pxc-pre-upgrade.py"]
+          env:
+            - name: STATEFULSET_NAME
+              value: pxc-backup-mongodb
+            - name: NAMESPACE
+              value: {{ .Release.Namespace }}  
+            - name: TARGET_VERSION
+              value: {{ .Values.images.mongodbImage.tag }}
+            - name: K8S_SERVICE_NAME
+              value: pxc-backup-mongodb-headless
+            - name: MONGODB_PORT
+              value: "27017"
+            - name: MONGODB_ROOT_USERNAME
+              value: root
+            - name: MONGODB_ROOT_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  key: mongodb-root-password
+                  name: pxc-backup-mongodb
+            - name: MYSQL_STATEFULSET_NAME
+              value: pxcentral-mysql
+            - name: MYSQL_ROOT_USERNAME
+              value: root
+            - name: MYSQL_ROOT_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  key: DB_PASSWORD
+                  name: pxcentral-mysql-secret
+            - name: JOB_TIMEOUT_SECONDS
+              value: "600"
+            - name: MONGODB_IMAGES
+              value: >
+                {
+                  "registry": "{{ .Values.mongodbImages.registry }}",
+                  "repo": "{{ .Values.mongodbImages.repo }}",
+                  "imageName": "{{ .Values.mongodbImages.imageName }}",
+                  "tags": {
+                    "mongodb5": "{{ .Values.mongodbImages.tags.mongodb5 }}",
+                    "mongodb6": "{{ .Values.mongodbImages.tags.mongodb6 }}",
+                    "mongodb7": "{{ .Values.mongodbImages.tags.mongodb7 }}"
+                  }
+                }
+          volumeMounts:
+            - name: log-volume
+              mountPath: /logs
+            - name: dump-volume
+              mountPath: /dumps
+      volumes:
+        - name: log-volume
+          persistentVolumeClaim:
+            claimName: pre-upgrade-log-pvc
+        - name: dump-volume
+          persistentVolumeClaim:
+            claimName: pre-upgrade-dump-pvc
+      restartPolicy: Never
+  backoffLimit: 0
```

Comment:
@t-nidhin we try to perform a rollback to the initial version in case of any failure, a retry might not be appropriate here as the state/version of the mongodb deployment may be in an intermediate state.